test/python/circuit/classical/test_expr_constructors.py

```diff
@@ -388,16 +482,29 @@ def test_index_explicit(self):
             expr.index(cr, 3),
             expr.Index(expr.Var(cr, types.Uint(4)), expr.Value(3, types.Uint(2)), types.Bool()),
         )
+        self.assertFalse(expr.index(cr, 3).const)
+
         self.assertEqual(
             expr.index(a, cr),
             expr.Index(a, expr.Var(cr, types.Uint(4)), types.Bool()),
         )
+        self.assertFalse(expr.index(a, cr).const)
+
+        self.assertEqual(
+            expr.index(255, 1),
+            expr.Index(expr.Value(255, types.Uint(8)), expr.Value(1, types.Uint(1)), types.Bool()),
+        )
+        self.assertTrue(expr.index(255, 1).const)
 
     def test_index_forbidden(self):
         with self.assertRaisesRegex(TypeError, "invalid types"):
             expr.index(Clbit(), 3)
         with self.assertRaisesRegex(TypeError, "invalid types"):
             expr.index(ClassicalRegister(3, "a"), False)
+        with self.assertRaisesRegex(TypeError, "invalid types"):
+            expr.index(ClassicalRegister(3, "a"), 1.0)
+        with self.assertRaisesRegex(TypeError, "invalid types"):
+            expr.index(0xFFFF, 1.0)
```

Comment:
Can  you add one that asserts it's invalid to index into a float with an integer too?